### PR TITLE
Project list view

### PR DESCRIPTION
**Pull request by Marc-Week**
Created test case for rendering project list, couple of TODO for error handling, Will need to test completely with "gather list items" story.

### CODE REVIEW

File: routes/projects.js [hunk 1]
```diff
@@ -3,7 +3,8 @@ let router = express.Router();
 
 /* GET home page. */
 router.get('/', function (req, res, next) {
-  res.render('projects', { title: 'Jama Software Capstone', projects: req.session.projects });
+  res.status(200);
```
Comment (chances) [inline routes/projects.js]: 200 is the default `responseCode` when using render. This is superfluous.

File: test/routes/projects.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/* eslint-env mocha */
+
+let chai = require('chai');
+let expect = chai.expect;
+let chaiHttp = require('chai-http');
+let server = require('../../app');
+let ejs = require('ejs');
+let read = require('fs').readFileSync;
+let join = require('path').join;
+chai.use(chaiHttp);
```
Comment (chances) [inline test/routes/projects.js]: Pad this with a blank line above

File: test/routes/projects.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+          expect(res).to.have.status(200);
+          expect(res).to.redirect();
+          expect(res).to.have.property('text');
+          expect(res.charset).to.equal('utf-8');
+          path = join(__dirname, '../../views/projects.ejs');
+            // TODO need to test with actuall data once "Gather projects" story is completed
+          data = {title: 'Projects', projects: null};
+          renderedView = ejs.compile(read(path, 'utf8'), {filename: path})(data);
+          expect(res.text).to.equal(renderedView);
+          expect(res.text).contains('</ul>');
```
Comment (chances) [inline test/routes/projects.js]: Why is it expecting this?

File: views/projects.ejs [hunk 4]
```diff
@@ -1,16 +1,19 @@
...
-<!--TODO: add error messages -->
-<!-- TODO: On a post to this page, don't change the title of the page to the error message -->
-<script src="js/project.js"></script>
-</body>
+  </head>
+  <body>
+    <h1><%= title %></h1>
+    <h3>Click on a project below to view it's traceability graph:</h3>
+    <ul id="projectList">
+     <% if(projects != null){%>
```
Comment (chances) [inline views/projects.ejs]: Format this as if it was being linted. :wink:

File: views/projects.ejs [hunk 5]
```diff
@@ -1,16 +1,19 @@
...
+      <%projects.forEach(function(project){ %>
+       <li><a href="/graph?id=<%= project.id %>" data-id="<%= project.id %>"><%= project.name %></a></li>
+     <% }); } else{%>
+      <p>Sorry, there are no projects to display.</p><% }%>
```
Comment (chances) [inline views/projects.ejs]: Close the else block on its own line.

File: test/routes/projects.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        .post('/')
+        .send({teamName: 'sevensource', username: 'dummy', password: 'password'})
+        .end(function (err, res) {
+          if (err) {
+            console.log(err);
+          }
+          expect(res).to.have.status(200);
+          expect(res).to.redirect();
+          expect(res).to.have.property('text');
+          expect(res.charset).to.equal('utf-8');
```
Comment (chances) [inline test/routes/projects.js]: When wouldn't the response use this character encoding?

File: routes/projects.js (hunk 1 above)
Comment (ibilal) [inline routes/projects.js]: Okay I thought we had to explicitly set these. I'll take it out.

File: test/routes/projects.js (hunk 2 above)
Comment (ibilal) [inline test/routes/projects.js]: Above eslint-env mocha ?

File: test/routes/projects.js (hunk 3 above)
Comment (ibilal) [inline test/routes/projects.js]: Umm yeah. I'll take it out.

File: test/routes/projects.js (hunk 6 above)
Comment (ibilal) [inline test/routes/projects.js]: It wouldn't?

File: test/routes/projects.js (hunk 2 above)
Comment (ibilal) [inline test/routes/projects.js]: chai.use(chaiHttp)?

**Comment by chances** [comment]
LGTM

**Pull request merged**